modules/variables/main.tf

```diff
@@ -0,0 +1,10 @@
+resource "tfe_variable" "this" {
+  for_each = var.variables
+
+  category     = var.category
+  hcl          = var.hcl
+  key          = var.category == "env" ? "TF_VAR_${each.key}" : each.key
+  sensitive    = var.sensitive
+  value        = var.hcl ? replace(replace(jsonencode(each.value), "/(\".*?\"):/", "$1 = "), "/= null/", "= \"\"") : tostring(each.value)
```

Comment:
do we really want to map `null` to `""`? That seems wrong and many of our modules need `null`

Reply:
Correct, for `var.category == "env"` then we probably want `null` mapped to `""`